common/persistence/execution_manager_test.go

```diff
@@ -0,0 +1,242 @@
+		{
+			method: "CreateFailoverMarkerTasks",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().CreateFailoverMarkerTasks(gomock.Any(), gomock.Any()).Return(nil)
+			},
+		},
+		{
+			method: "GetTimerIndexTasks",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().GetTimerIndexTasks(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "CompleteTimerTask",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().CompleteTimerTask(gomock.Any(), gomock.Any()).Return(nil)
+			},
+		},
+		{
+			method: "RangeCompleteTimerTask",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().RangeCompleteTimerTask(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "CompleteReplicationTask",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().CompleteReplicationTask(gomock.Any(), gomock.Any()).Return(nil)
+			},
+		},
+		{
+			method: "RangeCompleteReplicationTask",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().RangeCompleteReplicationTask(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "DeleteWorkflowExecution",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().DeleteWorkflowExecution(gomock.Any(), gomock.Any()).Return(nil)
+			},
+		},
+		{
+			method: "DeleteCurrentWorkflowExecution",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().DeleteCurrentWorkflowExecution(gomock.Any(), gomock.Any()).Return(nil)
+			},
+		},
+		{
+			method: "GetCurrentExecution",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().GetCurrentExecution(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "ListCurrentExecutions",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().ListCurrentExecutions(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "IsWorkflowExecutionExists",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().IsWorkflowExecutionExists(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "GetReplicationDLQSize",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().GetReplicationDLQSize(gomock.Any(), gomock.Any()).Return(nil, nil)
+			},
+		},
+		{
+			method: "GetReplicationTasks",
+			prepareMocks: func(mockedStore *MockExecutionStore) {
+				mockedStore.EXPECT().GetReplicationTasks(gomock.Any(), gomock.Any()).Return(&InternalGetReplicationTasksResponse{
+					Tasks: []*InternalReplicationTaskInfo{
+						{
+							DomainID: "test",
+							TaskID:   1,
+						},
+						{
+							DomainID: "test",
+							TaskID:   2,
+						},
+					},
+					NextPageToken: nil,
+				}, nil)
+			},
+			verifyRes: func(t *testing.T, resValues []reflect.Value) {
```

Comment:
This one is very different from the others and also makes you to introduce/check for `verifyRes`. Is it possible to extract it to a separate tests? The rest is super-intuitive and look alike.